Multiple-source RPQ example

bahbyega opened the pull request:
Переписан пример 8.2.1 RPQ с несколькими стартовыми #17

bahbyega committed:
Переписан пример 8.2.1 RPQ с несколькими стартовыми

gsvgit commented on tex/RPQ.tex:
```diff
@@ -12,197 +12,438 @@ \section{Достижимость с несколькими источникам
+
+Для синхронизации обхода составим набор блочно--диагональных матриц, каждая из которых --- это прямая сумма двух матриц:
+$D_{0\_a} = R_{0\_a} \oplus G_{0\_a}$ и $D_{0\_a} = R_{0\_b} \oplus G_{0\_b}$.
+
```
А можете эти матрицы сделать блочными, как, например, в главе 11.

gsvgit commented on tex/RPQ.tex:
```diff
@@ -12,197 +12,438 @@ \section{Достижимость с несколькими источникам
+
+Сложим два полученных вектора, чтобы получить новый фронт: \fbox{0 1 0} \fbox{0 1 0 0} + \fbox{1 0 0} \fbox{0 0 0 1} = \fbox{1 1 0} \fbox{0 1 0 1}.
+
+То есть в наш фронт \fbox{0 1 0 1} попали вершины 1 и 3 соотвественно. А именно, мы совершили следующие переходы в графе.
```
Было бы здорово рядом с графом такую же картинку для автомата иметь.

gsvgit commented on tex/RPQ.tex:
```diff
@@ -12,197 +12,438 @@ \section{Достижимость с несколькими источникам
+Покажем на примере, как данный метод может быть использован, когда мы накладываем дополнительные ограничения в виде регулярного языка на путь в графе. 
+
+Для этого, во-первых, предъявим булевые представления для матриц смежности графа и автомата для регулярного языка. Затем, введем специальную блочно--диагональную матрицу 
+для синхронизации обхода в ширину по двум матрицам смежности. Далее, попробуем наивно реализовать обход в ширину, и посмотрим, почему наивная реализация
```
Думается, где-то тут надо пояснить про классический алгоритм пересечения автоматов, чтобы было понятно, что за синхронизация.

gsvgit commented on tex/RPQ.tex:
```diff
@@ -12,197 +12,438 @@ \section{Достижимость с несколькими источникам
+    \end{matrix}
+\end{alignat*}
+
+Сложим два полученных вектора, чтобы получить новый фронт: \fbox{0 1 0} \fbox{0 1 0 0} + \fbox{1 0 0} \fbox{0 0 0 1} = \fbox{1 1 0} \fbox{0 1 0 1}.
```
Вот тут вот первый вектор разрывается на два куска. Не самая удачная вёрстка. Может тожно как-то без fbox? В 13.2, вроде, иначе свёрстано.

gsvgit commented on tex/RPQ.tex:
```diff
@@ -12,197 +12,438 @@ \section{Достижимость с несколькими источникам
+
+\begin{alignat*}{7}
+  b:\,\,
+  & \begin{matrix}
```
Думаю, лучше операцию умноения таки явно писать, не опускать.

gsvgit commented on tex/RPQ.tex:
```diff
@@ -12,197 +12,438 @@ \section{Достижимость с несколькими источникам
+Его матрица смежности имеет следующий вид.
+\[ G_1 =
+\begin{pmatrix}
+. & [a] & . & [b] \\
```
Для множеств честные фигурные скобки

gsvgit commented on tex/RPQ.tex:
```diff
@@ -12,197 +12,438 @@ \section{Достижимость с несколькими источникам
+  \end{tikzpicture}
+\end{center}
+
+Таким образом, в ответ попадает вершина 2.
```
Что-то картинка, кажется, точно такая же, как в изначальном примере с проблемой.

bahbyega committed:
Исправления в примере 8.2.1

- матрицы сделаны блочными
- к картинке обхода графа добавлена картинка для автомата
- fbox используется теперь только в math mode
- явно указана операция умножения между векторами и матрицами
- в картинки с последним шагом обхода добавлен зеленый цвет для выделения результата
